src/event.rs

```diff
@@ -225,107 +139,61 @@ pub enum WindowEvent<'a> {
-    ReceivedCharacter(char),
+    CharReceived(char),
 
-    /// The window gained or lost focus.
-    ///
-    /// The parameter is true if the window has gained focus, and false if it has lost focus.
-    Focused(bool),
-
-    /// An event from the keyboard has been received.
-    KeyboardInput {
-        device_id: DeviceId,
-        input: KeyboardInput,
-        /// If `true`, the event was generated synthetically by winit
-        /// in one of the following circumstances:
-        ///
-        /// * Synthetic key press events are generated for all keys pressed
-        ///   when a window gains focus. Likewise, synthetic key release events
-        ///   are generated for all keys pressed when a window goes out of focus.
-        ///   ***Currently, this is only functional on X11 and Windows***
-        ///
-        /// Otherwise, this value is always `false`.
-        is_synthetic: bool,
-    },
+    KeyPress(KeyPress),
 
     /// The keyboard modifiers have changed.
     ///
+    /// This is tracked internally to avoid tracking errors arising from modifier key state changes when events from
+    /// this device are not being delivered to the application, e.g. due to keyboard focus being elsewhere.
+    ///
     /// Platform-specific behavior:
     /// - **Web**: This API is currently unimplemented on the web. This isn't by design - it's an
     ///   issue, and it should get fixed - but it's the current state of the API.
     ModifiersChanged(ModifiersState),
 
-    /// The cursor has moved on the window.
-    CursorMoved {
-        device_id: DeviceId,
+    PointerCreated(PointerId),
+    PointerForce(PointerId, Force),
+    PointerMoved(PointerId, PhysicalPosition<f64>),
+    PointerPress(PointerId, PointerPress),
+    PointerEntered(PointerId),
+    PointerLeft(PointerId),
+    PointerDestroyed(PointerId),
 
-        /// (x,y) coords in pixels relative to the top-left corner of the window. Because the range of this data is
-        /// limited by the display area and it may have been transformed by the OS to implement effects such as cursor
-        /// acceleration, it should not be used to implement non-cursor-like interactions such as 3D camera control.
-        position: PhysicalPosition<f64>,
-        #[deprecated = "Deprecated in favor of WindowEvent::ModifiersChanged"]
-        modifiers: ModifiersState,
-    },
-
-    /// The cursor has entered the window.
-    CursorEntered { device_id: DeviceId },
-
-    /// The cursor has left the window.
-    CursorLeft { device_id: DeviceId },
-
-    /// A mouse wheel movement or touchpad scroll occurred.
-    MouseWheel {
-        device_id: DeviceId,
-        delta: MouseScrollDelta,
-        phase: TouchPhase,
-        #[deprecated = "Deprecated in favor of WindowEvent::ModifiersChanged"]
-        modifiers: ModifiersState,
-    },
+    ScrollStarted,
+    ScrollLines(UnitlessDelta<f64>),
+    ScrollPixels(PhysicalDelta<f64>),
+    ScrollEnded,
```

Comment:
I also wonder the idea behind event for start/stop scrolling, seems like events for ScrooLine and ScrollPixels are already enough.

Reply:
We exposed a [`TouchPhase` field on the old `MouseWheel` event](https://docs.rs/winit/0.22.2/winit/event/enum.WindowEvent.html#variant.MouseWheel.field.phase) and `ScrollStarted`/`ScrollEnded` allow us to expose that functionality in the new API.

Reply:
I feel like the naming/docs are pretty confusing about those. The docs are saying that they are about touchscreen, in reality they are used for some mouse/touchpad scroll as well. Also old API is using `TouchPhase::Ended`, when you've removed all your fingers from the touch surface, so using scroll here is just confusing. If we take into account the other `Pointer` events handling of touch is just more confusing at this point, unless you want to force users to track touch phase themself.

Reply:
I agree that implying that the scroll phase and the touchscreen touch phase are related is confusing, why is why this redesign doesn't do that. `TouchPhase` in `MouseWheel` was [originally exposed in order to implement overscrolling on macOS](https://github.com/rust-windowing/winit/commit/7886ff9acdbc4961359725536c1a15e2cbb3d4f5), and this design should let us continue to do that. The exact API guarantees around when `ScrollStarted` and `ScrollEnded` get emitted should be specified in the docs, though.